### PR TITLE
Introduce config library

### PR DESCRIPTION
**Pull request by timja**
Required for adding tests in https://github.com/jenkins-infra/account-app/pull/167.

As integration tests fail due to no config.properties existing.
I could create one on CI but it'd just be a pain for everyone all round.

Let's use an actual modern framework that supports env variables natively.

Library is https://github.com/lightbend/config

cc @NotMyFault @halkeye 

Fixes https://github.com/jenkins-infra/account-app/issues/162

Means we don't hit stupid issues like https://github.com/jenkins-infra/account-app/pull/159 again. Where an 'example' file is actually relied on in production

### CODE REVIEW

**Comment by halkeye** [comment]
i like the idea of cleanness

Since its account app, I think @wadeck should also review new dependencies.

Otherwise looks good to me

File: README.md [hunk 1]
```diff
@@ -1,117 +1,65 @@
...
-(As you can see above, this connects your test instance to the actual LDAP
-server, so the data you'll be seeing is real.
-
-The command line system properties are for JIRA LDAP sync tool. JIRA user account you are providing has to have the system admin access to JIRA.
-TODO: feed this data from config.properties
-
-### Docker Compose
-
-A docker compose file can be used for testing purpose.
+and port forward to the production ldap instance:
```
Comment (dduportal) [inline README.md:30]: This is a recipe for disaster 😱 

Is there a way to use the docker-ldap image instead to provide a working environment?

**Review by dduportal — APPROVED** [review]
Nice job! Let's plan the deployment (once reviewed) when the team will have some time this week

File: README.md (hunk 1 above)
Comment (timja) [reply README.md:30]: This is what the README previously documented just with the old setup.

I only maintained the docs for the new setup, I’m happy to remove it not attached to it at all

**Review by Wadeck — APPROVED** [review]
The new dependency seems to be fine.

**Review by NotMyFault — APPROVED** [review]

**Comment by dduportal** [comment]
I'm gonna proceed to merge, the infra team will take over for testing the deployment with these changes.

File: README.md (hunk 1 above)
Comment (timja) [reply README.md:30]: the docs I added also reference using mock-ldap which is a working ldap container and the expected default dev environment

**Pull request merged**